lib/rubocop/cop/layout/rescue_ensure_alignment.rb

```diff
@@ -29,13 +29,25 @@ class RescueEnsureAlignment < Cop
               '%<begin_loc_line>d, %<begin_loc_column>d.'.freeze
         ANCESTOR_TYPES = %i[kwbegin def defs class module].freeze
         RUBY_2_5_ANCESTOR_TYPES = (ANCESTOR_TYPES + [:block]).freeze
+        ANCESTOR_TYPES_WITH_ACCESS_MODIFIERS = %i[def defs].freeze
 
         def on_resbody(node)
-          check(node) unless modifier?(node)
+          check node unless modifier? node
         end
 
         def on_ensure(node)
-          check(node)
+          check node
+        end
+
+        def autocorrect(node)
+          whitespace = whitespace_range node
+          return false unless whitespace.source.strip.empty?
```

Comment:
It isn't immediately clear to me what code would cause `return false unless whitespace.source.strip.empty?` or `return false if alignment_node.nil?` to trigger. Is there a way for us to be able to handle auto-correcting in any of these scenarios?

Reply:
`alignment_node.nil?` We couldn't find a node for alignment. It has very low probability, but I couldn't add this check.

`whitespace.source.strip.empty?` is an old code. It looks strange. I've found that it means `whitespace.real_whitespace?`. You can reproduce that some whitespace could not be a real whitespace =)

```
new_source = autocorrect_source(<<-RUBY.strip_indent)
  def test
    'foo'; rescue StandardError;
  end
RUBY

expect(new_source).to eq(<<-RUBY.strip_indent)
  def test
    'foo'; rescue StandardError;
  end
RUBY
```

Reply:
I will try to fix it and add some tests.